packages/frontend/src/redux/actions/staking.js

```diff
@@ -533,3 +524,85 @@ export const handleUpdateCurrent = (accountId) => async (dispatch, getState) =>
     let currentAccount = selectStakingCurrentAccountbyAccountId(getState(), { accountId });
     dispatch(staking.updateCurrent({ currentAccount }));
 };
+
+export const getValidatorFarmData = (validator, accountId) => async (dispatch, getState) => {
```

Comment:
**Not a blocker for this PR**, but...

1. This thunk should be setting loading state and error states so that the UI doesn't show a 0 balance in case
2. Directly related to the lack of loading and error states managed here, we should also not be blindly swallowing _all_ errors encountered from `get_unclaimed_reward` and returning a `0` value for balance in those cases, which would be misleading to the user.

@esaminu please make sure we've got this issue documented so we can folllow up and fix this up so there is user feedback when something has gone wrong instead of them seeing a 0 balance.